libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Views/SFSDKWindowManager.m

```diff
@@ -0,0 +1,308 @@
+ SalesforceSDKCore
+ 
+ Created by Raj Rao on 7/4/17.
+ 
+ Copyright (c) 2017-present, salesforce.com, inc. All rights reserved.
+ 
+ Redistribution and use of this software in source and binary forms, with or without modification,
+ are permitted provided that the following conditions are met:
+ * Redistributions of source code must retain the above copyright notice, this list of conditions
+ and the following disclaimer.
+ * Redistributions in binary form must reproduce the above copyright notice, this list of
+ conditions and the following disclaimer in the documentation and/or other materials provided
+ with the distribution.
+ * Neither the name of salesforce.com, inc. nor the names of its contributors may be used to
+ endorse or promote products derived from this software without specific prior written
+ permission of salesforce.com, inc.
+ 
+ THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR
+ IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND
+ FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+ CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+ WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY
+ WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+#import "SFSDKWindowManager.h"
+#import "SFSDKWindowContainer.h"
+#import "SFSDKRootController.h"
+#import "SFApplicationHelper.h"
+#import "SFSecurityLockout.h"
+@interface SFSDKWindowManager() <SFSDKWindowContainerDelegate> {
+    SFSDKWindowContainer *_prevActiveWindow;
+    SFSDKWindowContainer *_currentWindow;
+}
+@property (nonatomic, strong) NSMutableOrderedSet *delegates;
+
+@end
+
+@implementation SFSDKWindowManager
+
+static const CGFloat SFWindowLevelAuthOffset      = 100;
+static const CGFloat SFWindowLevelPasscodeOffset  = 200;
+static const CGFloat SFWindowLevelSnapshotOffset  = 2000;
+static NSString *const kSFMainWindowKey     = @"main";
+static NSString *const kSFLoginWindowKey    = @"auth";
+static NSString *const kSFSnaphotWindowKey  = @"snapshot";
+static NSString *const kSFPasscodeWindowKey = @"passcode";
+
+- (instancetype)init {
+    
+    self = [super init];
+    if (self) {
+        _namedWindows = [NSMapTable mapTableWithKeyOptions:NSMapTableStrongMemory
+                                              valueOptions:NSMapTableStrongMemory];
+        _delegates = [NSMutableOrderedSet orderedSet];
+    }
+    return self;
+    
+}
+
+- (SFSDKWindowContainer *)mainWindow {
+    if (![self.namedWindows objectForKey:kSFMainWindowKey]) {
+        [self setMainUIWindow:[SFApplicationHelper sharedApplication].windows[0]];
```

Comment:
I'd suggest that this logic also include the possibility of the `UIApplicationDelegate.window` property.  So something like:
```obj-c
if ([SFApplicationHelper sharedApplication].delegate.window != nil) {
    [self setMainUIWindow:[SFApplicationHelper sharedApplication].delegate.window];
} else if ([SFApplicationHelper sharedApplication].windows[0] != nil) {
    [self setMainUIWindow:[SFApplicationHelper sharedApplication].windows[0]];
} else {
    // We should probably consider an adverse case where there are no windows defined.
}
```